### DISCUSSION TITLE
Map "Detected Language" property to Modern PNP Search

### DISCUSSION THREAD

**Discussion by ValiPrk** [microsoft-search/pnp-modern-search#3520] [General]
Hello,

In the Enterprise advanced search there is an option called "only the language(s)" and I have been trying to replicate this option by adding a filter to Modern PNP Filter webpart with the crawled property "ows_q_TEXT_TranslationLanguage", but unfortunately I don't receive anything, only "null" outcomes.
Is there a way to implement this?
![Languages](https://github.com/microsoft-search/pnp-modern-search/assets/147488275/d751947e-8120-450b-afdb-daf46133d947)

**Comment by kasperbolarsen** [comment]
I tried a variant of this scenario: I set up my intranet site collection (happens to be the root as weel) to work with 3 of the languages I know, English, German and Danish.
I added a News Post and added two translated version as per https://support.microsoft.com/en-gb/office/create-multilingual-sharepoint-sites-pages-and-news-2bb7d610-5453-41c6-a0e8-6f40b3ed750c

After the indexer picked it up, I now see 3 hits in search, one for each language, and the  SPTranslationLanguage contains the expected value:

![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/a3c5792c-632b-4a5a-aed3-90bfd7d111a3)


I hope this will be of use for you :-)

**Reply by ValiPrk** [reply]
Hi @kasperbolarsen, Thank you for your reply, I'm testing the solution right now.
I have one more question, does it also apply to Office documents (Word, Excel...)?

**Reply by kasperbolarsen** [reply]
As far as I know the Out Of The Box answer is no, as SharePoint doesn't offer any on the fly translation of office documents. However, you could extend the document metadata (Content type + Site Column) with a Term based field identifying the language of the document.

**Reply by ValiPrk** [reply]
Maybe "OWS_DOCUMENTLANGUAGE" would work, have you tried it?

**Reply by kasperbolarsen** [reply]
I have no such property in any of my tenants. Are you sure it is an out of the box property?

**Reply by ValiPrk** [reply]
Hi @kasperbolarsen, I checked and indeed, the "SPTranslationLanguage" property is only related to pages and we needed something for all the documents in our tenant.
Until Microsoft will come with a solution that can predict automatically the language from the text of a document, we will use your suggestion -> (Content type + Site Column) with a Term based field identifying the language of the document.

Thank you very much for your help!

**Comment by wobba** [comment]
You can use the managed properties `language` and `languages` to work with language of any file. This is populated during indexing, not in SharePoint, and is using some logic to determine the languages of the content. Not always accurate.

**Comment by kasperbolarsen** [comment]
Looks like this is about to become a lot more common,  https://techcommunity.microsoft.com/t5/sharepoint-premium-blog/sharepoint-premium-translation-now-generally-available/ba-p/4055393
